ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostAdapter.java

```diff
@@ -176,6 +175,7 @@ public void showInterstitial() {
 
     @Override
     public void onDestroy() {
+        ChartboostSingleton.detachBanners();
```

Comment:
Are you sure about this? Seems like when a banner adapter is asked to be destroyed you're removing all the other banners.
Isn't a ChartboostAdapter created for each banner request?

Reply:
One way to fix this would be to keep a reference to the banner inside the adapter and not the singleton. IF a ChartboostAdapter is instantiated for each request. Would also make it more modular.

Reply:
I think this can be reused and create multiple banners within one request so I made it the way that singleton holds every banner reference and when reach on destroy (cause the whole view is dead at this point and gonna have to go through oncreate) we should detach all the banners (most likely only one).

Reply:
I see two issues here. First, we need to confirm that assumption that the ChartboostAdapter can be reused for multiple requests. I know that's not the case in iOS. Second, even if ChartboostAdapter can be reused, if it is possible that more than one is created for different requests, then this line of code will break all the adapters.